ci2/ci/github.py

```diff
@@ -271,14 +274,42 @@ def __init__(self, branch):
         self.branch = branch
         self.prs = None
         self.sha = None
-        self.healing = False
-        self.changed = False
+        self.updating = False
+        self.github_changed = True
+        self.batch_changed = True
+
+    async def notify_github_changed(self, app):
+        self.github_changed = True
+        await self._update(app)
+
+    async def notify_batch_changed(self, app):
+        self.batch_changed = True
+        await self._update(app)
 
     async def update(self, app):
+        # update everything
+        self.github_changed = True
+        self.batch_changed = True
+        await self._update(app)
+
+    async def _update(self, app):
+        if self.updating:
+            return
+        self.updating = True
```

Comment:
shouldn't this be an atomically acquired lock (and if it's not available we just return)? Right now two concurrent threads of control could try cancelling the same batch job (which I guess is fine? but will trigger 400s in heal)

Reply:
and maybe that supersedes the repos_lock because only one thread is updating at a time?

Reply:
There should only be one thread, so this is atomic.

Reply:
Since there is no `await`.  Clarify of asyncio for the win.